src/QsCompiler/DataStructures/Diagnostics.fs

```diff
@@ -465,23 +465,24 @@ type DiagnosticItem =
         with
         | _ -> str // let's fail silently for now
 
-    static member Message(code: ErrorCode, args: IEnumerable<string>) =
-        let unlines = String.concat Environment.NewLine
+    static member Message(code, args) =
+        let typeMismatch summary expected actual =
+            String.concat Environment.NewLine [ summary; "Expected: " + expected; "  Actual: " + actual ]
 
         let message =
             match code with
```

Comment:
The numbering for the arguments in these error messages seems pretty out of order. Is this does to maintain consistency with how these error codes would have been pass arguments before?

Reply:
I wanted to use the same function `Diagnostic.typeContextArgs` to create the argument list for all the type errors, and let each error message decide the best order to display them in. Unfortunately it does look pretty confusing with the argument numbers - this would be cleaner if #1025 was resolved and each diagnostic argument had a name instead of a number.